tools/ClangPlugin.cpp

```diff
@@ -169,7 +174,9 @@ namespace clad {
         // If this is the last required derivative order, replace the function
         // inside a call to clad::differentiate/gradient with its derivative.
         if (request.CallUpdateRequired && lastDerivativeOrder)
-          request.updateCall(DerivativeDecl, m_CI.getSema());
+          request.updateCall(DerivativeDecl,
```

Comment:
I think we can avoid sending an extra argument to `updateCall` by deciding here only which of `DerivativeDecl` and `OverloadedDerivativeDecl` should be `replacementFD`. 

```suggestion
          request.updateCall((OverloadedDerivativeDecl ? OverloadedDerivativeDecl
                                                 : DerivativeDecl),
                       m_CI.getSema());
```

Reply:
The extra argument is also required for `dump` ing the actual gradient function. Also when Hessian Mode uses the gradient it can use the actual gradient instead of the overloaded gradient.

Reply:
Oh okay, I didn't consider these things. Thanks for the details.